nomad/state/state_store.go

```diff
@@ -226,12 +226,11 @@ func (s *StateStore) UpdateNodeStatus(index uint64, nodeID, status string) error
 			return fmt.Errorf("error retrieving any allocations for the node: %v", nodeID)
 		}
 		for _, alloc := range allocs {
-			copyAlloc := new(structs.Allocation)
-			*copyAlloc = *alloc
+			copyAlloc := alloc.Copy()
 			if alloc.ClientStatus == structs.AllocClientStatusPending ||
 				alloc.ClientStatus == structs.AllocClientStatusRunning {
 				copyAlloc.ClientStatus = structs.AllocClientStatusLost
-				if err := s.updateSummaryWithAlloc(index, copyAlloc, watcher, txn); err != nil {
+				if err := s.updateSummaryWithAlloc(index, copyAlloc, alloc, watcher, txn); err != nil {
```

Comment:
Put a comment explaining why we mark the allocation as lost and why we have to update the summary